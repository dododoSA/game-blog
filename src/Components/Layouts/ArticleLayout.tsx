import styles from 'styles/Layouts/ArticleLayout.module.scss';
import { SideCard } from 'Components/SideContents/SideCard';

// 記事関連のページのレイアウト
const ArticleLayout = ({ children }: { children: React.ReactNode }) => (
  <div className={styles.container}>
    <div className={styles.article_wrapper}>
      <article className={styles.main_content}>{children}</article>
      <aside className={styles.side_bar}>
        <SideCard>
          <p>hello</p>
        </SideCard>
      </aside>
    </div>
    <div>recent posts etc.</div>
  </div>
);

export { ArticleLayout };
