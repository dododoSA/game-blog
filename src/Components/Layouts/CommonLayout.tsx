import { BreadcrumbNavigation } from 'Components/Common/BreadcrumbNavigation/BreadcrumbNavigation';
import { Header } from 'Components/Common/Header/Header';
import { Footer } from 'Components/Common/Footer/Footer';
import styles from 'styles/Layouts/CommonLayout.module.scss';

// 全ページ共通のレイアウト
const CommonLayout = ({ children }: { children: React.ReactNode }) => (
  <>
    <Header />
    <main>
      <div className={styles.container}>
        {children}
      </div>
    </main>
    <Footer />
  </>
);

export { CommonLayout };
