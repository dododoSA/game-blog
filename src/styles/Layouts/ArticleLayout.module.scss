.container {
  padding: 0 4rem;
  margin: 0 auto;
  max-width: 1200px;

  .article_wrapper {
    display: flex;
    justify-content: space-between;

    .main_content {
      width: calc(100% - 330px);

    }

    .side_bar {
      width: 300px;

    }

  }
}

// TODO: mixin
// https://zenn.dev/web_tips/articles/095d7419507ca9

@media screen and (max-width: 992px){
  .container {
    padding: 0 2.5rem;
    > .article_wrapper {
      > .main_content {
        width: 100%;
      }

      > .side_bar {
        display: none;
      }
    }
  }
}
