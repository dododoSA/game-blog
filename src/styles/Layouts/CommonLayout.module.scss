.container {
  // TODO: 背景色の設定箇所がここでいいのかどうか後で確認する
  background-color: #e9e9e9;
}