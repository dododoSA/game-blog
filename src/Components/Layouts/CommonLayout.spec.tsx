import renderer from 'react-test-renderer';
import { CommonLayout } from 'Components/Layouts/CommonLayout';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      // TODO: 他のパターン
      pathname: '/',
    };
  },
}));

test('CommonLayout', () => {
  const component = renderer.create(<CommonLayout>hello</CommonLayout>);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
