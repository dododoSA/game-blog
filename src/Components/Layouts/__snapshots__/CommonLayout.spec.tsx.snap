// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonLayout 1`] = `
Array [
  <header>
    <div
      className="logo_wrapper"
    >
      <div
        className="container"
      >
        <a
          className="logo"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Rのゲームプログラミング教室
        </a>
      </div>
    </div>
    <div>
      <nav
        className="nav_bar"
      >
        <ul>
          <li>
            <div
              className="container"
            >
              <a
                className="menu_item"
                href="/theory"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                理論編
              </a>
              <ul>
                <li>
                  <a
                    href="/theory/pglanguage"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                  >
                    プログラミング言語
                  </a>
                  <ul>
                    <li>
                      <a
                        href="/theory/pglanguage/c"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                      >
                        C言語
                      </a>
                    </li>
                  </ul>
                </li>
                <li>
                  <a
                    href="/theory/collision"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                  >
                    当たり判定
                  </a>
                </li>
              </ul>
            </div>
          </li>
          <li>
            <div
              className="container"
            >
              <a
                className="menu_item"
                href="/practice"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                実践編
              </a>
            </div>
          </li>
          <li>
            <div
              className="container"
            >
              <a
                className="menu_item"
                href="/about"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                About
              </a>
            </div>
          </li>
        </ul>
      </nav>
    </div>
  </header>,
  <main>
    <div
      className="container"
    >
      hello
    </div>
  </main>,
  <footer>
    <div
      className="copyright"
    >
      ©︎2021 R
    </div>
  </footer>,
]
`;
